tests/test_traits.py

```diff
@@ -1001,3 +1001,29 @@ def test_modify_traits():
 
     with pytest.raises(VerifyException, match="Nope"):
         op.verify()
+
+
+def test_modify_parent_traits():
+    """
+    Test that adding traits do not modify parent traits when they are inherited.
+    """
+
+    class ParentOp(IRDLOperation, ABC):
+        name = "test.parent"
+
+        traits = traits_def(LargerOperandTrait())
+
+    @irdl_op_definition
+    class ModifiedOp(ParentOp):
+        name = "test.modified"
+
+    @irdl_op_definition
+    class NotModifiedOp(ParentOp):
+        name = "test.not_modified"
+
+    ModifiedOp.traits.add_trait(BitwidthSumLessThanTrait(64))
```

Comment:
Random thought, should this be the API instead?
```suggestion
    ModifiedOp.add_trait(BitwidthSumLessThanTrait(64))
```

Reply:
I agree `traits.add_trait` seems kind of redundant?